File: JVM/jnetcall-java-common/src/main/java/jnetcall/java/impl/util/ClassTools.java
package jnetcall.java.impl.util;

import java.lang.reflect.Method;

import org.javatuples.Pair;

import jnetbase.java.meta.Reflect;
import jnetcall.java.api.flow.MethodCall;

public final class ClassTools {

    private static final String AsyncAdd = "_A";

    public static Pair<String, String> toMethodId(MethodCall call) {
        String name = call.M().replace("_", "");
        int count = call.A().length;
        String id = (name + "_" + count).toLowerCase();
        String ida = (id + AsyncAdd).toLowerCase();
        return Pair.with(ida, id);
    }

    public static String toMethodId(Method method) {
        String name = method.getName().replace("_", "");
        int count = method.getParameters().length;
        String suffix = Reflect.isAsync(method) ? AsyncAdd : "";
        String id = (name + "_" + count + suffix).toLowerCase();
        return id;
    }

    public static boolean isSameMethod(Method m, String callName) {
        String mName = m.getName().replace("_", "");
        return mName.equalsIgnoreCase(callName);
    }

    public static String toDelegateId(Object del) {
        int hash = del.hashCode();
        Class type = del.getClass();
        Class interf = type.getInterfaces()[0];
        Method method = interf.getMethods()[0];
        return hash + "#" + method;
    }
}

File: NET/JNetCall.Sharp.Common/Impl/Util/ClassTools.cs
using System;
using System.Reflection;
using JNetBase.Sharp.Meta;
using JNetCall.Sharp.API.Flow;

namespace JNetCall.Sharp.Impl.Util
{
    public sealed class ClassTools
    {
        private const string AsyncAdd = "_A";

        public static (string, string) ToMethodId(MethodCall call)
        {
            var name = call.M.Replace("_", string.Empty);
            var count = call.A.Length;
            var id = (name + "_" + count).ToLowerInvariant();
            var ida = (id + AsyncAdd).ToLowerInvariant();
            return (ida, id);
        }

        public static string ToMethodId(MethodInfo method)
        {
            var name = method.Name.Replace("_", string.Empty);
            var count = method.GetParameters().Length;
            var suffix = Reflect.IsAsync(method) ? AsyncAdd : string.Empty;
            var id = (name + "_" + count + suffix).ToLowerInvariant();
            return id;
        }

        private const StringComparison Cmp = StringComparison.InvariantCultureIgnoreCase;

        public static bool IsSameMethod(MemberInfo m, string callName)
        {
            var mName = m.Name.Replace("_", string.Empty);
            return mName.Equals(callName, Cmp);
        }

        public static string ToDelegateId(Delegate del)
        {
            var hash = del.Target?.GetHashCode() ?? 0;
            var method = del.Method;
            return hash + "#" + method;
        }
    }
}